File: BotsSDK/korebotsdklib/src/main/java/kore/botssdk/models/BrandingStaticLinksModel.java
package kore.botssdk.models;

import kore.botssdk.utils.StringUtils;

public class BrandingStaticLinksModel {
    private Boolean show;
    private String layout;

    public BrandingStaticLinksModel updateWith(BrandingStaticLinksModel configModel) {
        show = configModel.show != null ? configModel.show : show;
        layout = !StringUtils.isNullOrEmpty(configModel.layout) ? configModel.layout : layout;
        return this;
    }
}

File: BotSdk/botclient/src/main/java/com/kore/network/api/responsemodels/branding/BrandingStaticLinksModel.kt
package com.kore.network.api.responsemodels.branding

import com.google.gson.annotations.SerializedName

data class BrandingStaticLinksModel(
    @SerializedName("show")
    val show: Boolean?,
    @SerializedName("layout")
    val layout: String = ""
)

fun BrandingStaticLinksModel.updateWith(configModel: BrandingStaticLinksModel): BrandingStaticLinksModel {
    return this.copy(
        show = configModel.show ?: show,
        layout = layout.ifEmpty { layout }
    )
}